### PR TITLE
feat(modules-sdk): Module as singleton instances

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
What:
    Medusa module's instance is initialized only once

At the moment only one instance per module is allowed, in the future if we need multiple instances of the same service we can hash the `declaration` property of the method to identify them.

### CODE REVIEW

**Review by fPolic — APPROVED** [review]
LGTM

File: packages/modules-sdk/src/loaders/__tests__/medusa-module.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  moduleLoader: jest.fn().mockImplementation((...args) => mockModuleLoader()),
+}))
+
+describe("Medusa Module", () => {
+  beforeEach(() => {
+    jest.resetModules()
+    jest.clearAllMocks()
+  })
+
+  it.only("MedusaModule bootstrap - Singleton instances", async () => {
```
Comment (olivermrbl) [inline packages/modules-sdk/src/loaders/__tests__/medusa-module.ts]: nit: rm
```suggestion
  it("MedusaModule bootstrap - Singleton instances", async () => {
```

File: packages/modules-sdk/src/loaders/__tests__/medusa-module.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        resolve: "@path",
+        options: {
+          abc: 123,
+        },
+      } as InternalModuleDeclaration,
+      {}
+    )
+
+    expect(mockRegisterMedusaModule).toBeCalledTimes(1)
+    expect(mockModuleLoader).toBeCalledTimes(1)
```
Comment (olivermrbl) [inline packages/modules-sdk/src/loaders/__tests__/medusa-module.ts:42]: question: Are these required? Shouldn't this test run the bootstrapping twice and make the assertion once?

**Review by olivermrbl — APPROVED** [review]
LGTM, with minor comments :)

File: packages/modules-sdk/src/loaders/__tests__/medusa-module.ts (hunk 2 above)
Comment (carlos-r-l-rodrigues) [reply packages/modules-sdk/src/loaders/__tests__/medusa-module.ts:42]: The test below is doing it, calling again the bootstrap and the calledTimes should continue to be 1.

Comment (olivermrbl) [reply packages/modules-sdk/src/loaders/__tests__/medusa-module.ts:42]: Yes, I understand. I would just think asserting only once, in the end, yields the same result. But no need to go back and forth about this.

Comment (carlos-r-l-rodrigues) [reply packages/modules-sdk/src/loaders/__tests__/medusa-module.ts:42]: I see, sorry, just wanted to be explicity in the test for who is reading.

**Pull request merged**